Modules/timemodule.c

```diff
@@ -2390,3 +2398,44 @@ pysleep(PyTime_t timeout)
     return -1;
 #endif
 }
+
+
+#ifndef MS_WINDOWS
+// time.sleep(0) optimized implementation.
+// On error, raise an exception and return -1.
+// On success, return 0.
+static int
+pysleep_zero_posix(void)
+{
+    static struct timeval zero = {0, 0};
```

Comment:
Hmm, why make this a static local instead of a const global?

Reply:
`select(2)` does not allow for a constant since it can modify the pointer (but it wouldn't in practice since both fields are 0).

Reply:
I'm not sure I trust that for all systems, we might end up with some interesting edge cases. It's probably safer to just make it a thread local global.

Reply:
I'll just make a local variable simply. That will reduce the diff and avoid bothering about portability (it eventually shouldn't really matter performance-wise probably)